src/PerfDetectiveExample/PDEmain.cpp

```diff
@@ -0,0 +1,46 @@
+#include "fpcsc/perf_util/sleep.h"
+
+#include <cstdlib>
+
+static int x2 = 0;
+
+int preCalculate(int z) {
+  if (x2 < 10) return 0;
+  return 1;
+}
```

Comment:
Might be optimized out, which can be prevented by declaring x2 volatile. Do you plan to also check here how instrumentation behaves under optimization?